### ISSUE TITLE
Is this only extractive? 

### ISSUE THREAD

**Issue by Hellisotherpeople** [Huffon/sentence-compressor#1]
It looks to be the holy grail of what I've always looked for - a grammatically correct word level highlighting tool. 

You've starred my CX_DB8 repo. Could I simply run your tool over each sentence for a better highlighting experience rather than my current method?

**Comment by Huffon** [comment]
Hi Allen,
Thank you for your interest!

First of all, there are two types of datasets in the Sentence Compression task.
One is extraction-based datasets you've mentioned,
And the other is abstraction-based datasets like [Gigaword](https://www.tensorflow.org/datasets/catalog/gigaword).

So If you train your compression model with only extraction-based datasets, 
I think yes, you can use the compression model as a high-lighting tool.

But as far as I know, there is not that much amount of extraction-based datasets.
And another concern is the compression model in this repository based on [BART](https://arxiv.org/abs/1910.13461) architecture.
So it might be too heavy to use it as a highlighting tool.

In [preprocess.py](https://github.com/Huffon/sentence-compressor/blob/master/utils/preprocess.py#L151), you can find extractive-based datasets except Gigaword.

**Comment by Hellisotherpeople** [comment]
I actually plan on making a new dataset with both word level extracts and abstracts from a diverse set of documents. Hopefully a conference will be ready to publish it because I will be submitting to recent NLP conferences. 

Is the pretrained BART model trained without Gigaword trained only on extractive datasets? Or do other abstractive datasets make it in to there? If its only trained on extractive datasets than I'm **super** excited and will try this out as soon as I can 

As far as highlighting goes, I'm okay if it's really compute intensive or GPU based. At this point, I'd only be using it for personal experiments and creating debate evidence for students who I coach. 

If I can get training working with this, I will contribute trained extractive "highlighter" models and likely be pointing folks who want a more sophisticated version of CX_DB8 to you because I've searched long and hard for a high quality modern solution to the highlighting issue and the only "advantage" that my method would offer is "biasable" summarization (summarization based on a query sentence) and not requiring additional training - but grammatical correctness is far more important.

**Comment by Huffon** [comment]
Yes. In the [Example](https://github.com/Huffon/sentence-compressor#example) section, the result of **BART (fine-tuned without Gigaword)** means that BART is only trained on an extract basis. data set.

I also uploaded [fine tuned version](https://drive.google.com/open?id=19C9zKJXq3vGZZ-9s-zi4Vi1ytWBzo_a_) of BART for your quick testing.
You can download and test it with your sentences.

At the moment, I think this is a PoC (Proof of Concept) level project.
And I will keep improving this elaborately :)

Hope this project helps with CX_DB8.
